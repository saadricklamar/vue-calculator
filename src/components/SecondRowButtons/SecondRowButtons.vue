<template>
  <section class="second-row-buttons">
    <button v-on:click="displayBtnValue($event)" class="seven" value="7">7</button>
    <button v-on:click="displayBtnValue($event)" class="eight" value="8">8</button>
    <button v-on:click="displayBtnValue($event)" class="nine" value="9">9</button>
    <button v-on:click="multiply($event)" class="multiply" value="x">x</button>
  </section>
</template>

<script>
export default {
  name: "SecondRowButtons",
  props: ["displayBtnValue", "multiply"]
};
</script>

<style>
p {
  color: white;
}
.multiply {
  background-color: #42b983;
  color: white;
}
</style>