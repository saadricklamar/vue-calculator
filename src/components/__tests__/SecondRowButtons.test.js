import { mount } from "@vue/test-utils";
import SecondRowButtons from "../SecondRowButtons/SecondRowButtons";

describe("SecondRowButtons", () => {
  const wrapper = mount(SecondRowButtons, {
    propsData: {
      displayBtnValue: jest.fn()
    }
  });
  it("should match the snapshot", () => {
    expect(wrapper.isVueInstance()).toMatchSnapshot();
  });
  it("should have the props of displayBtnValue", () => {
    expect(wrapper.props("displayBtnValue")).toBeTruthy();
  });
  it("should call displayBtnValue when 7 is clicked", () => {
    wrapper.vm.displayBtnValue = jest.fn();
    wrapper.find("button.seven").trigger("click");
    expect(wrapper.vm.displayBtnValue).toHaveBeenCalled();
  });
  it("should call displayBtnValue when 8 is clicked", () => {
    wrapper.vm.displayBtnValue = jest.fn();
    wrapper.find("button.eight").trigger("click");
    expect(wrapper.vm.displayBtnValue).toHaveBeenCalled();
  });
  it("should call displayBtnValue when 9 is clicked", () => {
    wrapper.vm.displayBtnValue = jest.fn();
    wrapper.find("button.nine").trigger("click");
    expect(wrapper.vm.displayBtnValue).toHaveBeenCalled();
  });
});
