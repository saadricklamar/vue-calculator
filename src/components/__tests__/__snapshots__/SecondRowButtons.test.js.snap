// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondRowButtons should match the snapshot 1`] = `true`;
